File: warmup_1/Not_string.py
'''
Given a string, return a new string where "not " has been added to the front.
However, if the string already begins with "not", return the string unchanged.
not_string('candy') → 'not candy'
not_string('x') → 'not x'
not_string('not bad') → 'not bad'

author LZ-FSDev
see https://codingbat.com/prob/p189441
'''

def not_string(str):
    if (str[0:3] != "not"):
        str = "not " + str
    return str

print("not_string('candy') -> " + not_string('candy'))
print("not_string('x') -> " + not_string('x'))
print("not_string('not bad') -> " + not_string('not bad'))

File: warmup_1/NotString.java
/**
 * Given a string, return a new string where "not " has been added to the
 * front. However, if the string already begins with "not", return the string
 * unchanged. Note: use .equals() to compare 2 strings.
 * 
 * <ul>
 *  <li>notString("candy") → "not candy"
 *  <li>notString("x") → "not x"
 *  <li>notString("not bad") → "not bad"
 * </ul>
 * 
 * @author LZ-FSDev
 * @see https://codingbat.com/prob/p191914
 * @since 17.0.1
 * @version 0.0.1
 */
public class NotString {
    public static void main(String[] args) {
        System.out.println("notString(\"candy\") -> " + notString("candy"));
        System.out.println("notString(\"x\") -> " + notString("x"));
        System.out.println("notString(\"not bad\") -> " + notString("not bad"));
    }

    /**
     * Given a string, return a new string where "not " has been added to the
     * front. However, if the string already begins with "not", return the
     * string unchanged. Note: use .equals() to compare 2 strings.
     * 
     * @param str A string.
     * @return str with "not" at the beginning.
     * @since 0.0.1
     */
    public static String notString(String str) {
        if (str.length() < 3 || !str.substring(0, 3).equals("not")) {
            str = "not " + str;
        }
        return str;
    }
}
